statsmodels/survey/summary_stats.py

```diff
@@ -0,0 +1,612 @@
+        # but if you're within the cluster to be removed, set as 0
+        w[self.sclust == c] = 0
+        return w
+
+    def _bootstrap_weights(self):
+        """
+        Computes bootstrap replicate weight
+
+        Returns
+        -------
+        w : ndarray
+            Augmented weight
+        """
+        w = self.weights.copy()
+        clust_count = np.zeros(self.nclust)
+        for s in range(self.nstrat):
+            # how to handle strata w/ only one cluster?
+            w[self.strat == s] *= float(self.ncs[s] - 1) \
+                                         / self.ncs[s]
+            # If there is only one cluster then weights wont change
+            if self.ncs[s] == 1:
+                continue
+
+            # resample array of clusters
+            ii_resample = np.random.choice(self.ii[s], size=(self.ncs[s]-1))
+            # accumulate number of times cluster i was resampled
+            clust_count += np.bincount(ii_resample,
+                                       minlength=max(self.sclust)+1)
+
+        w *= clust_count[self.sclust]
+        return w
+
+    def _mean_bootstrap_weight(self, bsn):
+        """
+        Computes mean bootstrap replicate weight
+
+        Parameters
+        ----------
+        bsn : integer
+            Mean bootstrap averages the number of resampled clusters over bsn
+        Returns
+        -------
+        w : ndarray
+            Augmented weight
+        """
+        clust_count = np.zeros(self.design.nclust)
+        w = self.weights.copy()
+        # for each replicate, I accumulate bsn number of times?
+        for b in range(bsn):
+            for s in range(self.nstrat):
+                w[self.strat == s] *= ((float(self.ncs[s] - 1) /
+                                        self.ncs[s])**(1/bsn))
+                # If there is only one or two clusters then weights wont change
+                if (self.ncs[s] == 1 or self.ncs[s] == 2):
+                    continue
+                # resample array of clusters in strata s
+                ii_resample = np.random.choice(self.ii[s], size=(self.ncs[s]-1))
+                # accumulate number of times cluster i was resampled
+                clust_count += np.bincount(ii_resample,
+                                           minlength=max(self.sclust)+1)
+        # avg number of times cluster i was resampled
+        clust_count /= bsn
+        # augment weights
+        w *= clust_count[self.sclust]
+        return w
+
+
+class SurveyStat(SurveyDesign):
+    """
+    Estimation and inference for summary statistics in complex surveys.
+
+    Parameters
+    -------
+    design : SurveyDesign object
+
+    Attributes
+    ----------
+    est : ndarray
+        The point estimates of the statistic, calculated on the columns
+        of data.
+    vcov : ndarray
+        The variance-covariance of the estimates.
+    pseudo : ndarray
+        The jackknife pseudo-values.
+    """
+
+    def __init__(self, design, mse=False):
```

Comment:
Curious, which circumstances promote this idea for the user instead of just specifying mse or not. Also, I'm not sure what you mean by 'centering around the estimator' for the jackknife mean, as that is just the average of all the 'delete 1' estimates for the statistic you're calculating